screen/contributors/src/main/java/com/ivy/contributors/IvyWalletRepositoryDataSource.kt

```diff
@@ -38,19 +39,48 @@ class IvyWalletRepositoryDataSource @Inject constructor(
 
     companion object {
         private const val CONTRIBUTORS_PER_PAGE = 100
+        private const val DISPLAY_ANONYMOUS_CONTRIBUTORS = true
+        private const val INITIAL_PAGE = 1
     }
 
-    suspend fun fetchContributors(): List<ContributorDto>? {
-        return try {
-            withContext(Dispatchers.IO) {
+    suspend fun fetchContributors(): List<ContributorDto> {
+        return withContext(Dispatchers.IO) {
+            val contributors = mutableListOf<ContributorDto>()
+            paging(addContributors = { results ->
+                results?.forEach { contributor -> contributors.add(contributor) }
+            })
+            contributors.toList()
+        }
+    }
+
+    private suspend fun paging(addContributors: (List<ContributorDto>?) -> Unit) {
+        var currentPage: Int? = INITIAL_PAGE
+        while (currentPage != null) {
+            val contributors = try {
                 httpClient
                     .get("https://api.github.com/repos/Ivy-Apps/ivy-wallet/contributors") {
+                        parameter("anon", DISPLAY_ANONYMOUS_CONTRIBUTORS)
```

Comment:
Let's extract the HTTP request in another function and make it return `Either`

Reply:
What you think about this? :

```
suspend fun fetchContributors(): Result<List<ContributorDto>> {
        return withContext(dispatchersProvider.io) {
            try {
                val contributors = mutableListOf<ContributorDto>()
                paging(addContributors = { results ->
                    results?.forEach { contributor -> contributors.add(contributor) }
                })
                Result.success(contributors)
            } catch (e: Exception) {
                Timber.tag("FETCH_CONTRIBUTORS_PAGING").d(e)
                Result.failure(e)
            }
        }
    }

private suspend fun paging(addContributors: (List<ContributorDto>?) -> Unit) {
        var currentPage: Int? = INITIAL_PAGE
        while (currentPage != null) {
            val contributors = getContributorsFromRequest(currentPage)
            currentPage = getNextPage(contributors.getOrThrow(), currentPage)
            currentPage?.let { addContributors(contributors.getOrThrow()) }
        }
    }

private suspend fun getContributorsFromRequest(currentPage: Int): Result<List<ContributorDto>> =
        try {
            val contributors = httpClient
                .get("https://api.github.com/repos/Ivy-Apps/ivy-wallet/contributors") {
                    parameter("anon", DISPLAY_ANONYMOUS_CONTRIBUTORS)
                    parameter("per_page", CONTRIBUTORS_PER_PAGE)
                    parameter("page", currentPage)
                }
                .body<List<ContributorDto>>()
            Result.success(contributors)
        } catch (e: Exception) {
            Timber.tag("FETCH_CONTRIBUTORS_REQUEST").d(e)
            Result.failure(e)
        }

```

So the viewmodel will call `fetchContributors` and perform it in this way: 

```
val contributorsResult = ivyWalletRepositoryDataSource.fetchContributors()

contributorsResult
            .onSuccess { contributorsDto ->
                val contributors = contributorsDto.map {
                    Contributor(
                        name = it.login ?: "Anonymous",
                        photoUrl = it.avatarUrl ?: "",
                        contributionsCount = it.contributions.toString(),
                        githubProfileUrl = it.link ?: ""
                    )
                }
                contributorsResponse.value = ContributorsResponse.Success(
                    contributors.toImmutableList()
                )
            }
            .onFailure { throwable ->
                contributorsResponse.value = ContributorsResponse.Error(
                    throwable.message.toString()
                )
            }
``

Reply:
Can you change the code so I can comment 😅 LGTM, a few small nits:
- We use `Either` from Arrow instead of `Result` (check the dev guidelines for the reasoning)
- You can contributorsResult.fold()
- Don't throw exceptions
- Can we use `addAll` here to simplify it? 1 results?.forEach { contributor -> contributors.add(contributor)`

Reply:
@rodrigoliveirac check this out https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/guidelines/Error-Handling.md

The benefit of Either is that we're not limited to `Throwable` error type

Reply:
> Can you change the code so I can comment 😅 LGTM, a few small nits:
> 
> * We use `Either` from Arrow instead of `Result` (check the dev guidelines for the reasoning)
> * You can contributorsResult.fold()
> * Don't throw exceptions
> * Can we use `addAll` here to simplify it? 1 results?.forEach { contributor -> contributors.add(contributor)`

Sure! I have a question about another topic: Detekt throws an error about the default values that I put in ContributorDto because the 'Anonymous' one comes null. What's the best approach to resolve that?

Reply:
It's an failure in our Detekt check, default values are fine for DTOs, use `Suppress`

Reply:
Ok. Thanks!